server/src/library/externalLibraryUtils.ts

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as PQLS from "@microsoft/powerquery-language-services";
+
+export type ExternalSymbolLibrary = ReadonlyArray<PQLS.LibrarySymbol.LibrarySymbol>;
+export type IncomingExternalSymbolLibrary = ExternalSymbolLibrary | undefined | null;
```

Comment:
I wanted the contract to use an explicit `null` value to indicate that the symbol file was removed/no longer needed. I removed the `undefined` from the type annotation but left the check in the code since the value comes in from a remote caller and there is no guarantee that its well-formed.